### PR TITLE
Add rotate! and reflect! functions

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

**Review by jpsamaroo — APPROVED** [review]
Looks good!

File: test/rocarray/blas.jl [hunk 1]
```diff
@@ -21,6 +21,36 @@ end
...
+                x = rand(T, 8)
+                y = rand(T, 8)
+                Rx = ROCArray(x)
+                Ry = ROCArray(y)
+                c = rand(cty)
+                s = rand(T)
+                rotate!(Rx, Ry, c, s)
+                @test isapprox(c*x + s*y, Array(Rx))
+                @test isapprox(-conj(s)*x + c*y, Array(Ry))
+            end
```
[the remark this answers is not in the document]
Comment (amontoison) [reply test/rocarray/blas.jl:36]: `c` is always real because reflection matrices are Hermitian and `c` is a coefficient of the diagonal.

**Comment by amontoison** [comment]
off topic: @jpsamaroo 
Do you know if it's possible to add a generic fallback for `reflect!` in `GPUArrays.jl`?
I don't know how to implement it without a kernel.

**Comment by pxl-th** [comment]
Superseded by #484

**Pull request closed without merging**